datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -282,15 +284,21 @@ protected void rewindAndPausePartitionOnException(TopicPartition srcTopicPartiti
   }
 
   protected void rewindAndPausePartitionsOnSendException() {
-    // For all topic partitions which have seen send exceptions, attempt to rewind them to the last checkpoint
+    // For all topic partitions which have seen send exceptions, attempt to rewind them to the last checkpoint.
+    // The outcome of the rewind can fall into three categories:
+    // 1) The rewind is successful and the Exception returned by the SendCallback is transient. This TopicPartition is
+    //    not added to the auto-pause list.
+    // 2) The rewind is successful and the Exception returned by the SendCallback is non-transient. This TopicPartition
+    //    is added to the auto-pause list with SEND_ERROR as the reason.
+    // 3) The rewind itself failed. An exception is thrown, and the connector task is brought down to avoid wrongfully
```

Comment:
nit: do you need both ',' and 'and'

Reply:
also, can you rephrase this comment may be: the connector task is brought down to avoid committing wrong checkpoints.

Reply:
done